File: 2020/2020W06/PreppindataW06.py
import pandas as pd
import re

xlsx = pd.ExcelFile(r'E:/PD 2020 Wk 6 Input.xlsx')
conv = pd.read_excel(xlsx, 'GBP to USD conversion rate')
conv['Week'] = conv['Date'].apply(lambda x: 'wk '+str(int(x.strftime('%U'))+1)+x.strftime(' %Y'))
conv['Conv Rate'] = conv['British Pound to US Dollar'].apply(lambda x: float(re.search('(?<=\s)(\d\.\d+)',x).group(1)))
conv = conv.groupby('Week',as_index=False).agg({'Conv Rate':[('Max Conv Rate','max'),('Min Conv Rate','min')]})
conv.columns = [col[1] if col[1] else col[0] for col in conv.columns.values]

sales = pd.read_excel(xlsx, 'Sales')
sales['Week'] = sales.apply(lambda x: 'wk '+str(x['Week'])+' '+str(x['Year']), axis=1)
sales['UK Sales Value (GBP)'] = sales['Sales Value']*(1-sales['US Stock sold (%)']/100)
sales['US Sales Value (GBP)'] = sales['Sales Value']-sales['UK Sales Value (GBP)']

final = pd.merge(sales,conv,how='inner',on='Week')
final['US Sales (USD) Best Case'] = final['US Sales Value (GBP)']*final['Max Conv Rate']
final['US Sales (USD) Worst Case'] = final['US Sales Value (GBP)']*final['Min Conv Rate']
final['US Sales Potential Variance'] = final['US Sales (USD) Best Case']-final['US Sales (USD) Worst Case']
final = final[['Week', 'UK Sales Value (GBP)', 'US Sales (USD) Best Case', 'US Sales (USD) Worst Case', 'US Sales Potential Variance']].copy()

File: 2020/2020W06/preppindataw06.R
library(dplyr)
library(readxl)
library(stringr)

xlsx <- "E:/PD 2020 Wk 6 Input.xlsx"

final <- read_excel(xlsx, sheet = "GBP to USD conversion rate", col_types = c("date", "text")) %>%
  mutate('Week' = paste('wk',as.integer(strftime(`Date`,format='%U'))+1,strftime(`Date`,format='%Y')),
         'Conv Rate' = as.numeric(str_extract(`British Pound to US Dollar`,'(?<=\\s)(\\d\\.\\d+)'))) %>%
  group_by(`Week`) %>%
  summarise('Max Conv Rate' = max(`Conv Rate`),
            'Min Conv Rate' = min(`Conv Rate`)) %>%
  merge(.,read_excel(xlsx,sheet = "Sales") %>%
          mutate('Week' = paste('wk',`Week`,`Year`)), by='Week') %>%
  mutate('UK Sales Value (GBP)' = round(`Sales Value`*(1-`US Stock sold (%)`/100),2),
         'US Sales Value (GBP)' = `Sales Value`-`UK Sales Value (GBP)`,
         'US Sales (USD) Best Case' = round(`US Sales Value (GBP)`*`Max Conv Rate`,2),
         'US Sales (USD) Worst Case' = round(`US Sales Value (GBP)`*`Min Conv Rate`,2),
         'US Sales Potential Variance' = `US Sales (USD) Best Case`-`US Sales (USD) Worst Case`) %>%
  select(c('Week', 'UK Sales Value (GBP)', 'US Sales (USD) Best Case', 
           'US Sales (USD) Worst Case', 'US Sales Potential Variance'))

View(final)